_posts/plotting-overview.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title:  "Altair vs Matplotlib Plot Construction Overview"
+date:   2018-08-15 14:50:00 -0500
+author: "Kimberly Orr"
+categories: about
+tags: "intro about matplotlib altair"
+excerpt_separator: <!--read more-->
+---
+## Altair
+Altair uses a declarative grammar. As its [overview](https://altair-viz.github.io/getting_started/overview.html) states:
+>The key idea is that you are declaring links between data columns and visual encoding channels, such as the x-axis, y-axis, color, etc.As the overview states, "the key idea is that you are declaring links between data columns and visual encoding channels, such as the x-axis, y-axis, color, etc."
+
+So, the general process for creating a basic Altair plot is to specify your data:
+ ```python
+alt.Chart(df)
+ ``` 
+Specify what type of glyph/mark should be used to represent your data:
+```python
+mark_point()
+```
+Then, link your data columns with the encoding channels:
+ ```python
+encode(x=alt.X("column1"), y=alt.Y("column2"))
+ ```
+ 
+So that a finished plot would look like:
+ ```python
+# import
+import altair as alt
+# plot
+alt.Chart(df).mark_point().encode(
+    x=alt.X("column1"), y=alt.Y("column2"), color=alt.Color("column3")
+)
+```
+## Matplotlib
+Matplotlib is a powerful object-oriented procedural plotting library. Instead of linking data with encoding channels, Matplotlib uses an object-oriented interface to place objects on a canvas.
+
+The general thought process is to create a plot:
+```python
+fig, ax = plt.subplots()
+```
+Add a scatter plot to the axes object of this figure:
+```python
+ax.scatter(x_array, y_array)
```

Comment:
I think both the Altair and mpl examples should be runnable, so I think you should make a teensy dataframe at the top to use with both examples. Something like 'df = pd.DataFrame({'x':[1,2,3,4,5], 'y':[4,5,6,7,8])' You can add a throwaway sentence about how we have data we want to plot and python offers us all the options.